phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/ui/SearchBox.java

```diff
@@ -1,24 +1,29 @@
 package org.phoenicis.javafx.views.mainwindow.ui;
 
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
 import org.phoenicis.javafx.views.common.ThemeManager;
 
 import javafx.beans.property.StringProperty;
 import javafx.scene.control.TextField;
 import javafx.scene.control.Button;
 import javafx.scene.layout.AnchorPane;
 
+import java.util.function.Consumer;
+
 public class SearchBox extends AnchorPane {
     private TextField searchField;
     private Button clearButton;
 
     private static final String cleanButtonIcon = "icons/mainwindow/general/edit-clear.png";
 
-    public SearchBox(ThemeManager themeManager) {
+    public SearchBox(ThemeManager themeManager, Consumer<String> onSearch, Runnable onClear) {
         this.searchField = new TextField();
 
         this.searchField.getStyleClass().add("searchBar");
         this.searchField.prefHeightProperty().bind(this.prefHeightProperty());
         this.searchField.prefWidthProperty().bind(this.prefWidthProperty());
+        this.searchField.setOnAction(event -> onSearch.accept(getText()));
```

Comment:
The problem with the delayed search is, that it only delays it, which results in the same amount of work to be done only at a later time. 
The "new" search works different from the old search.
The old search refreshed the complete view after a search term has been entered or a category was selected. The new search does this through bindings. Therefore we set the applications/categories only once. Afterwards the search is done by applying a filter to the observable list, that contains the applications/categories. 
As far as I know this leads to a specific addition or removal of items and not to a rebuilding of the complete view.